starboard/nplb/posix_compliance/posix_time_clock_gettime_test.cc

```
@@ -0,0 +1,365 @@
+  const int kTrials = 100;
+  struct timespec ts_prev, ts_curr;
+
+  memset(&ts_prev, 0, sizeof(ts_prev));
+  errno = 0;
+  int ret_initial = clock_gettime(CLOCK_MONOTONIC_RAW, &ts_prev);
+  if (ret_initial == -1 && errno == EINVAL) {
+    GTEST_SKIP() << "CLOCK_MONOTONIC_RAW defined but not supported "
+                    "(returned EINVAL).";
+    return;
+  }
+  ASSERT_EQ(0, ret_initial)
+      << "Initial clock_gettime(CLOCK_MONOTONIC_RAW) failed. errno: " << errno;
+
+  int64_t previousTime = TimespecToMicroseconds(ts_prev);
+
+  for (int trial = 0; trial < kTrials; ++trial) {
+    memset(&ts_curr, 0, sizeof(ts_curr));
+    errno = 0;
+    ASSERT_EQ(0, clock_gettime(CLOCK_MONOTONIC_RAW, &ts_curr))
+        << "clock_gettime(CLOCK_MONOTONIC_RAW) in loop failed for trial "
+        << trial << ". errno: " << errno;
+    int64_t currentTime = TimespecToMicroseconds(ts_curr);
+    EXPECT_GE(currentTime, previousTime)
+        << "CLOCK_MONOTONIC_RAW time should be non-decreasing. Previous: "
+        << previousTime << ", Current: " << currentTime << " on trial "
+        << trial;
+    if (currentTime < previousTime) {
+      FAIL() << "CLOCK_MONOTONIC_RAW time DECREASED. Previous: " << previousTime
+             << ", Current: " << currentTime << " on trial " << trial;
+      return;
+    }
+    previousTime = currentTime;
+  }
+  SUCCEED();
+}
+#else   // CLOCK_MONOTONIC_RAW
+TEST(PosixTimeClockGettimeTests, ClockMonotonicRawNotTestedDueToMissingDefine) {
+  GTEST_SKIP() << "CLOCK_MONOTONIC_RAW not defined, skipping test.";
+}
+#endif  // CLOCK_MONOTONIC_RAW
+
+#ifdef CLOCK_REALTIME_COARSE
+TEST(PosixTimeClockGettimeTests, ClockRealtimeCoarseSucceeds) {
+  struct timespec ts;
+  memset(&ts, 0, sizeof(ts));
+  errno = 0;
+  int ret = clock_gettime(CLOCK_REALTIME_COARSE, &ts);
+  if (ret == -1 && errno == EINVAL) {
+    GTEST_SKIP() << "CLOCK_REALTIME_COARSE defined but not supported "
+                    "(returned EINVAL).";
+    return;
```

unnecessary (I also see other such cases below, but I'll stop repeating this comment)

---

It is necessary to ensure that the call for unsupported clocks don't cause a crash. Reporting it as a skipped tests vs silently succeeding is useful information about the system when the test results are reported.

In the update with TEST_P sections, changed to silently succeed the available test in this case, but remains logged as skipped tests for the increasing and monotonic tests.